cypress/commons/actions/generic/DatasetManager.js

```diff
@@ -90,13 +98,39 @@ export const getDatasetOverviewPlaceholderRetryButton = () => cy.get(SELECTORS.o
 export const getDatasetOverviewPlaceholderRollbackButton = () => cy.get(SELECTORS.overview.placeholder.rollbackButton);
 export const getDatasetOverviewPlaceholderApiLink = () => cy.get(SELECTORS.overview.placeholder.apiLink);
 
+export const getCategoryAccordionSummary = (categoryId) =>
+  cy.get(SELECTORS.overview.categories.accordionSummary.replace('$CATEGORY_ID', categoryId));
+export const getCategoryAccordionDetails = (categoryId) =>
+  cy.get(SELECTORS.overview.categories.accordionDetails.replace('$CATEGORY_ID', categoryId));
 export const getIndicatorCard = (kpiId) => cy.get(SELECTORS.overview.indicators.cardByKpiId.replace('$KPI_ID', kpiId));
 export const getKpiLoading = (parent) => parent.find(SELECTORS.overview.indicators.kpiLoading);
+export const getCategoryName = (parent) => parent.find(SELECTORS.overview.categories.name);
+export const getCategoryType = (parent) => parent.find(SELECTORS.overview.categories.type);
+export const getCategoryDescription = (parent) => parent.find(SELECTORS.overview.categories.description);
+export const getCategoryKpi = (parent, kpiId) =>
+  parent.find(SELECTORS.overview.categories.kpi.replace('$KPI_ID', kpiId));
+export const getCategoryKpiLabel = (parent) => parent.find(SELECTORS.overview.categories.kpiLabel);
+export const getIndicatorKpiLabel = (parent) => parent.find(SELECTORS.overview.indicators.kpiLabel);
 export const getKpiValue = (parent, timeout) =>
   parent.find(SELECTORS.overview.indicators.kpiValue, timeout ? { timeout: timeout * 1000 } : undefined);
+export const getCategoryAttributes = (parent) => parent.find(SELECTORS.overview.categories.attributes);
+export const getCategoryDetailsDialog = () => cy.get(SELECTORS.overview.categories.detailsDialog);
+export const getCategoryDetailsDialogOpenButton = (parent) =>
+  parent.find(SELECTORS.overview.categories.detailsDialogOpenButton);
+export const getDetailsDialogCategoryName = () => cy.get(SELECTORS.overview.categories.detailsDialogCategoryName);
+export const getDetailsDialogDatasetName = () => cy.get(SELECTORS.overview.categories.detailsDialogDatasetName);
+export const getCategoryDetailsDialogCloseButton = () => cy.get(SELECTORS.overview.categories.detailsDialogCloseButton);
 export const getKpiError = (parent) => parent.find(SELECTORS.overview.indicators.kpiError);
 export const getKpiUnknownState = (parent) => parent.find(SELECTORS.overview.indicators.kpiUnknownState);
 
+export const openCategoryDetailsDialog = (categoryId, query) => {
+  const alias = api.interceptPostDatasetTwingraphQuery(query.results);
```

Comment:
If the function doesn't use the query object but only `query.results`, then maybe we should put directly `results` as function argument :slightly_smiling_face: